PWGHF/HFL/Tasks/taskSingleMuonSource.cxx

```diff
@@ -0,0 +1,250 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+//
+// \file taskSingleMuonSource.cxx
+// \brief Task used to seperate single muons source in Monte Carlo simulation.
+// \author Maolin Zhang <maolin.zhang@cern.ch>, CCNU
+
+#include "TMath.h"
+#include "TString.h"
+#include "TDatabasePDG.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/ASoAHelpers.h"
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/HistogramRegistry.h"
+#include "ReconstructionDataFormats/TrackFwd.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+using namespace o2;
+using namespace o2::aod;
+using namespace o2::framework;
+using MyCollisions = soa::Join<aod::Collisions, aod::McCollisionLabels, aod::EvSels>;
+using McMuons = soa::Join<aod::FwdTracks, aod::McFwdTrackLabels, aod::FwdTracksDCA>;
+
+namespace
+{
+enum ParticleType {
+  ParticleIsID = 0,
+  ParticleIsMuon,
+  ParticleIsSecondary,
+  ParticleHasLightParent,
+  ParticleHasCharmParent,
+  ParticleHasBeautyParent,
+  ParticleHasQuarkoniumParent,
+  ParticleHasTauParent
```

Comment:
Consider removing `Particle` from the names. It is redundant. Also please add comments documenting what individual values mean.